### PR TITLE
[gha] sticky forge comment

### PR DESCRIPTION
**Pull request by rustielin**
### Description

A few improvements to the Forge UX:
* Move the GHA comment creation to `run_forge.sh` script, since it serves for a pretty good command output as well. The comment contents can be redirected using env var `FORGE_COMMENT`. 
* Generate some useful pre-Forge output such as auto-refreshing dashboards and logs. Use it to generate a comment that gets posted right before tests run in `FORGE_PRE_COMMENT`
* Use https://github.com/marocchino/sticky-pull-request-comment to post and update a single Forge output comment on PRs, rather than spamming the PR each time. Use the action in `recreate` mode, which ensures that the Forge comment will always append to the bottom, and since it's deleted and re-created, will trigger a GH notification. This makes it easier to see what is exactly running and when (and what stage it's in), rather than needing to click into the Github checks UI and look through the running actions.
* Also write some better comments and reorganize some stuff in `run_forge.sh` since it's getting quite complicated. We should ideally move some of the o11y resource generation and regression checking / failure cases into the Forge test runner itself.
* Fail Forge with a different comment if perf regression detected

As a bonus: stop doing `set -x` for genesis. Especially since we tail the logs in Forge, the output can get quite verbose.

### Test Plan

Run the script locally to get the comment output. `FORGE_RUNNER_MODE=local ./testsuite/run_forge.sh`. Check all generated o11y links are good

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2166)

### CODE REVIEW

File: testsuite/run_forge.sh [hunk 1]
```diff
@@ -155,26 +147,44 @@ FORGE_END_TIME_MS="$(date '+%s')000"
...
 P99_LATENCY=$(cat $FORGE_REPORT | grep -oE '[0-9]+ ms p99 latency' | awk '{print $1}')
 if [ -n "$AVG_TPS" ]; then
     echo "AVG_TPS: ${AVG_TPS}"
     echo "forge_job_avg_tps {FORGE_CLUSTER_NAME=\"$FORGE_CLUSTER_NAME\",FORGE_NAMESPACE=\"$FORGE_NAMESPACE\",GITHUB_RUN_ID=\"$GITHUB_RUN_ID\"} $AVG_TPS" | curl -u "$PUSH_GATEWAY_USER:$PUSH_GATEWAY_PASSWORD" --data-binary @- ${PUSH_GATEWAY}/metrics/job/forge
     if [[ "$AVG_TPS" -lt "$TPS_THRESHOLD" ]]; then
         echo "(\!) AVG_TPS: ${avg_tps} < ${TPS_THRESHOLD} tps"
-	if [ "$FORGE_RUNNER_MODE" != "local" ]; then
-           FORGE_EXIT_CODE=1
+        if [ "$FORGE_RUNNER_MODE" != "local" ]; then
+            FORGE_EXIT_CODE=1
```
Comment (sitalkedia) [inline testsuite/run_forge.sh:187]: Can we add the comment in the report that the test failed because of TPS/latency regression?

**Comment by davidiw** [comment]
Hmm... I kinda like the reposting. Is it really that spammy? I guess if there are errors. I don't suppose there's an easy way to distinguish.

File: testsuite/run_forge.sh (hunk 1 above)
Comment (rustielin) [reply testsuite/run_forge.sh:262]: done. lmk what you think.

**Review by sitalkedia — APPROVED** [review]
LGTM

**Pull request merged**